graph/src/data/subgraph/mod.rs

```diff
@@ -812,10 +823,106 @@ impl<S, D, T> PartialEq for BaseSubgraphManifest<S, D, T> {
     }
 }
 
-pub type UnresolvedSubgraphManifest =
-    BaseSubgraphManifest<SchemaData, UnresolvedDataSource, UnresolvedDataSourceTemplate>;
+/// SubgraphManifest with IPFS links unresolved
+type UnresolvedSubgraphManifest =
+    BaseSubgraphManifest<UnresolvedSchema, UnresolvedDataSource, UnresolvedDataSourceTemplate>;
+
+/// SubgraphManifest validated with IPFS links resolved
 pub type SubgraphManifest = BaseSubgraphManifest<Schema, DataSource, DataSourceTemplate>;
 
+/// Unvalidated SubgraphManifest
+// TODO: Make the tuple fields private
```

Comment:
I'd rather not have `TODO`s checked into master. Why should this be private and what's keeping us from doing that?

Reply:
Removed and made the tuple value private